### PR TITLE
[Enhancement] /item command

### PR DESCRIPTION
**Pull request by jhonynet**
Details of this command are already explained in the issue.

https://github.com/MUnique/OpenMU/issues/161

### CODE REVIEW

File: src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
+﻿// <copyright file="CreateItemChatCommandPlugIn.cs" company="MUnique">
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+// </copyright>
+
+using System.Collections.Generic;
+using MUnique.OpenMU.DataModel.Configuration.Items;
```
Comment (sven-n) [inline src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs]: usings should be inside the namespace

File: src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+        private static Item CreateItem(Player player, Arguments arguments)
+        {
+            var item = new TemporaryItem();
+            var itemDefinition = player.GameContext.Configuration.Items.FirstOrDefault(def => def.Group == arguments.Group && def.Number == arguments.Number);
+            if (itemDefinition == null)
+            {
+                throw new ArgumentException($"[GM][/item] {arguments.Group} {arguments.Number} does not exists");
+            }
+            item.Definition = itemDefinition;
```
Comment (sven-n) [inline src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs]: please add one empty line here after the closing bracket

File: src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                throw new ArgumentException(Usage);
+            }
+
+            return new Arguments()
+            {
+                Group = byte.Parse(argumentsList.ElementAt(0)),
+                Number = short.Parse(argumentsList.ElementAt(1)),
+                Level = byte.Parse(argumentsList.ElementAtOrDefault(2) ?? "0"),
+                Exc = byte.Parse(argumentsList.ElementAtOrDefault(3) ?? "0"),
+                // Every excellent item has Skill
```
Comment (sven-n) [inline src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs]: Every excellent item has Skill, but only if a Skill is available in the ItemDefinition

File: src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                        .First(o => o.OptionType == ItemOptionTypes.Luck)
+                };
+                item.ItemOptions.Add(optionLink);
+            }
+
+            if (arguments.Exc > 0)
+            {
+                var excellentOptions = item.Definition.PossibleItemOptions.SelectMany(o => o.PossibleOptions)
+                    .Where(o => o.OptionType == ItemOptionTypes.Excellent)
+                    .Take(arguments.Exc);
```
Comment (sven-n) [inline src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs]: I think ```Take``` here isn't correct. The Parameter is probably meant as a bit-mask, like a flags enum?
If this is the case you'd need to do smth like this:
```
.Where(o => (o.Number & arguments.Exc) > 0)
```

File: src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                        ItemOption = option
+                    };
+                    item.ItemOptions.Add(optionLink);
+                });
+            }
+
+            return item;
+        }
+
+        private static Arguments MarshallArguments(List<string> argumentsList)
```
Comment (sven-n) [inline src/GameLogic/PlugIns/CreateItemChatCommandPlugIn.cs]: I'd name the method ```ParseArguments```

**Comment by jhonynet** [comment]
All fixed plus Codacy interface suggestion and code style

Regards!!

**Comment by sven-n** [comment]
Great, Thank you :)

**Pull request merged**